src/QueryBuilder/QueryBuilder/helpers/query.js

```diff
@@ -48,63 +48,66 @@ export const getTransformedValue = (val) => {
+  const operator = item.operator.current;
+  const value = item.value.current;
+
+  switch (operator) {
+    case OPERATORS.EQUAL:
+      queryOperand = { [field]: { $eq: value } };
+      break;
+    case OPERATORS.NOT_EQUAL:
+      queryOperand = { [field]: { $ne: value } };
+      break;
+    case OPERATORS.GREATER_THAN:
+      queryOperand = { [field]: { $gt: value } };
+      break;
+    case OPERATORS.GREATER_THAN_OR_EQUAL:
+      queryOperand = { [field]: { $gte: value } };
+      break;
+    case OPERATORS.LESS_THAN:
+      queryOperand = { [field]: { $lt: value } };
+      break;
+    case OPERATORS.LESS_THAN_OR_EQUAL:
+      queryOperand = { [field]: { $lte: value } };
+      break;
+    case OPERATORS.IN:
+      queryOperand = { [field]: { $in: getTransformedValue(value) } };
+      break;
+    case OPERATORS.NOT_IN:
+      queryOperand = { [field]: { $nin: getTransformedValue(value) } };
+      break;
+    case OPERATORS.STARTS_WITH:
+      queryOperand = { [field]: { $regex: new RegExp(`^${escapeRegex(value)}`).source } };
+      break;
+    case OPERATORS.CONTAINS:
+      queryOperand = { [field]: { $regex: new RegExp(escapeRegex(value)).source } };
+      break;
+    default:
+      break;
+  }
+
+  return queryOperand;
+};
+
 export const sourceToMongoQuery = (source) => {
   const query = {};
-  const andQuery = [];
-  let queryItem = {};
-
-  const escapeRegex = (value) => {
-    const escapedValue = value.replace(/[/\-\\^$*+?.()|[\]{}]/g, '\\$&');
-
-    return `${escapedValue}`;
-  };
-
-  source.forEach((item) => {
-    const field = item.field.current;
-    const operator = item.operator.current;
-    const value = item.value.current;
-
-    switch (operator) {
-      case OPERATORS.EQUAL:
-        queryItem = { [field]: { $eq: value } };
-        break;
-      case OPERATORS.NOT_EQUAL:
-        queryItem = { [field]: { $ne: value } };
-        break;
-      case OPERATORS.GREATER_THAN:
-        queryItem = { [field]: { $gt: value } };
-        break;
-      case OPERATORS.GREATER_THAN_OR_EQUAL:
-        queryItem = { [field]: { $gte: value } };
-        break;
-      case OPERATORS.LESS_THAN:
-        queryItem = { [field]: { $lt: value } };
-        break;
-      case OPERATORS.LESS_THAN_OR_EQUAL:
-        queryItem = { [field]: { $lte: value } };
-        break;
-      case OPERATORS.IN:
-        queryItem = { [field]: { $in: getTransformedValue(value) } };
-        break;
-      case OPERATORS.NOT_IN:
-        queryItem = { [field]: { $nin: getTransformedValue(value) } };
-        break;
-      case OPERATORS.STARTS_WITH:
-        queryItem = { [field]: { $regex: new RegExp(`^${escapeRegex(value)}`).source } };
-        break;
-      case OPERATORS.CONTAINS:
-        queryItem = { [field]: { $regex: new RegExp(escapeRegex(value)).source } };
-        break;
-      default:
-        break;
-    }
+  const boolOperator = source.find(item => Boolean(item.boolean.current))?.boolean.current;
```

Comment:
Looks interesting :)
Could you give me some examples of this checking?

Reply:
@mariia-aloshyna  Thank you for bringing this to our attention. We currently only support one type of operator for our queries, but we will support others in the future.
In the current case, we simply take the first element of the array (the lines of our queries) and find the first one in which the operator is present. if there is one, we set it for the entire query (since we can be sure that we currently support only one). 
This behavior will be changed in the near future. Perhaps I'll add a comment, since it really doesn't look obvious. Thank you!